### DISCUSSION TITLE
plot_data_plane  - failed to parse level string

### DISCUSSION THREAD

**Discussion by haksulee** [dtcenter/METplus#2787] [File I/O] [answered]
Hi METplus support:

I use metplus-5.1 docker/container and tried to do plot_data_plane with a GEFSv12 data file named as gep02.t00z.pgrb2.0p25 (zip file attached; also available at https://drive.google.com/drive/folders/1iVSAWBVb5Lsezf3A3dBZcFdDwx1Pt0EW), but got the error "failed to parse level string". 

root@ce35e181e596:/met/metplus-5.1/tutorial/regridding# plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="(0,*,*,*)";' -v 4

ERROR  : VarInfo::set_level_info_grib() - failed to parse level string '(0,*,*,*)'

I checked https://dtcenter.org/book/export/html/1595 and https://github.com/dtcenter/METplus/discussions/1245, but I wasn't able to find how to find a "level" value and "name" from the GEFSv12 grib2 file.

I also tried following commands but all didn't work:

plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMP_2maboveground"; level="(0,*,*)"; file_type=NETCDF_NCCF;' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX:2 m above ground"; level="TMAX:2 m above ground";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="TMAX:2 m above ground";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="TMAX";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="2 m above ground";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="2 m above ground:0-6 hour max";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX:2 m above ground"; level="0-6 hour max";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="0-6 hour max";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="0-6 hour max fcst";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="ENS=+2";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="0-6 hour max fcst:ENS=+2";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="d=2011102700";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="2";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="(*,*)";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="(*,*,*)";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="(0,*,*)";' -v 4
plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="(0,*,*,*)";' -v 4

Below are the results from wgrib2: 
wgrib2 gep02.t00z.pgrb2.0p25.f006
1:0:d=2011102700:TMP:2 m above ground:6 hour fcst:ENS=+2
2:172889:d=2011102700:TMAX:2 m above ground:0-6 hour max fcst:ENS=+2
3:345802:d=2011102700:TMIN:2 m above ground:0-6 hour min fcst:ENS=+2
4:518715:d=2011102700:HGT:highest tropospheric freezing level:6 hour fcst:ENS=+2
5:731459:d=2011102700:APCP:surface:0-6 hour acc fcst:ENS=+2

From the above, how can I know what is "name" and "level" to be used in plot_data_plane?

Thank you
Haksu

[gep02.t00z.pgrb2.0p25.zip](https://github.com/user-attachments/files/17738869/gep02.t00z.pgrb2.0p25.zip)

![plot_data_plane error](https://github.com/user-attachments/assets/2c71a383-5363-4ff0-9297-e5681013d84c)

**Comment by CPKalb** [comment, marked as answer]
Hi:

It looks like you are having trouble with the level string in MET in grib files.  The level string specified with parentheses and stars such as the example below is what is used for netCDF files.

```
level = (0,*,*)
```

For grib1/grib2 files, the level needs to be specified a different way.  [Section 5.2.11](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst) in the documentation explains the grib1/grib2 conventions.  In this case, you have to scroll down a little bit to get to the grib specific settings, so it can be tricky to find.  But, it's right under the "File-format specific settings for the “field” entry:" line.

I'm guessing you want to plot the variable TMAX that is record number 2 from the wgrib2 output?  If so, you will want to use the ZNNN setting for a vertical level.  The temperature is listed at 2m, so the level should be Z2.  I think your name and level should be set to the following:

```
'name="TMAX"; level="Z2";'
```

Does this work on your end?

Christina

**Reply by haksulee** [reply]
@CPKalb 

Thank you for the help.  
'name="TMAX"; level="Z2";'  worked ok and produced the map plot. 

plot_data_plane gep02.t00z.pgrb2.0p25.f006 TMP.gefs.ps 'name="TMAX"; level="Z2";' -v 4

![2mTmaxGEFSv12](https://github.com/user-attachments/assets/5a88d60d-9351-4496-9d50-e31ec132116c)

**Reply by CPKalb** [reply]
Great!  I'm going to close and lock this discussion.  If you have additional questions feel free to start up a new discussion.

Christina
